### PR TITLE
Azure IaC add new rego files (vm.rego , web.rego, aks.rego)

### PR DESCRIPTION
**Pull request by shahinshirvani**

### CODE REVIEW

File: azure/iac/aks.rego [hunk 1]
```diff
@@ -316,3 +316,111 @@ aks_aad_azure_rbac_metadata := {
...
+#
+# PR-AZR-0137-ARM
+#
+
+default aks_authorized_Ip = null
+
+azure_attribute_absence["aks_authorized_Ip"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.containerservice/managedclusters"
+    authorizedIPRange := resource.properties.apiServerAccessProfile.authorizedIPRanges[_]
```
Comment (rezoan) [inline azure/iac/aks.rego:330]: though its a array property in josn format checking like `not resource.properties.apiServerAccessProfile.authorizedIPRanges` should work as well.

File: azure/iac/aks.rego [hunk 2]
```diff
@@ -316,3 +316,111 @@ aks_aad_azure_rbac_metadata := {
...
+    not authorizedIPRange
+}
+
+azure_issue["aks_authorized_Ip"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.containerservice/managedclusters"
+    authorizedIPRange := resource.properties.apiServerAccessProfile.authorizedIPRanges[_]
```
Comment (rezoan) [inline azure/iac/aks.rego:337]: this should work as well `count(resource.properties.apiServerAccessProfile.authorizedIPRanges) == 0`

Comment (shahinshirvani) [reply azure/iac/aks.rego]: fixed

File: azure/iac/aks.rego (hunk 1 above)
Comment (shahinshirvani) [reply azure/iac/aks.rego]: fixed

**Review by rezoan — APPROVED** [review]

File: azure/iac/aks.rego [hunk 3]
```diff
@@ -316,3 +316,111 @@ aks_aad_azure_rbac_metadata := {
...
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.containerservice/managedclusters"
+    not resource.properties.networkProfile.networkPolicy
+}
+
+azure_issue["network_policy"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.containerservice/managedclusters"
+    lower(resource.properties.networkProfile.networkPolicy) != "azure"
+    lower(resource.properties.networkProfile.networkPolicy) != "calico"
```
[the remark this answers is not in the document]
Comment (farchide) [reply azure/iac/aks.rego:390]: @rezoan @shahinshirvani Calico is more complete than the Azure network policies for AKS. but in this specific test case, we are checking the existence of the network policy, either of them is ok

File: azure/iac/web.rego [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+} else = "Azure App Service Web app does not redirect HTTP to HTTPS" {
+    azure_issue["https_only"]
+}
+
+https_only_metadata := {
+    "Policy Code": "PR-AZR-0139-ARM",
+    "Type": "IaC",
+    "Product": "AZR",
+    "Language": "ARM template",
+    "Policy Title": "Ensure Azure App Service Web app redirects HTTP to HTTPS",
```
Comment (farchide) [inline azure/iac/web.rego:45]: @shahinshirvani 
Ensure Azure App Service Web App enforce https connection

**Pull request merged**